src/compile.rs

```diff
@@ -1504,17 +1535,37 @@ fn create_synthetic_resource(
         ));
     }
 
+    let class_name = if class.is_associated_resource(types) {
+        // TODO, this can be a lot cleaner if split *is* the check for .
+        class.name.as_ref().split_once('.').unwrap().1.into()
```

Comment:
If we don't have a `.` than this unwrap would be unsafe correct?  How sure are we that we will have a `.`

Reply:
`is_associated_resource()` does `.contains('.')`.  That says, I added the `basename` function that should be a lot cleaner here, rather than checking and splitting.  I'll update to that.